provider/blob/src/blob_schema.rs

```diff
@@ -80,8 +80,8 @@ impl<'data> BlobSchemaV1<'data> {
         let mut seen_min = false;
         let mut seen_max = false;
         for cursor in self.keys.iter0() {
-            for (_, idx) in cursor.iter1_copied() {
-                debug_assert!(idx < self.buffers.len());
+            for (locale, idx) in cursor.iter1_copied() {
+                debug_assert!(idx < self.buffers.len() || locale == Index32U8::SENTINEL);
```

Comment:
Issue: This is problematic because a newer blob won't pass the invariants of an older blob reader

Reply:
Personally ok with breaking debug assertions for new-blob-old-reader.

Reply:
We could also instead set the index to 0; it's fine because it'll never be loaded anyway (and if someone _does_ attempt to load it raw, they'll get a deserialize error or potentially GIGO behavior.

Reply:
The problem is that for a blob with just an empty key, 0 is also invalid as there will be zero buffers.

Reply:
(but also I'm not against breaking debug assertions with the =len sentinel in the first place